charmhelpers/contrib/openstack/templating.py

```diff
@@ -124,6 +127,20 @@ def complete_contexts(self):
         self.context()
         return self._complete_contexts
 
+    @property
+    def config_file(self):
+        return self._config_file
+
+    @property
+    def from_string(self):
```

Comment:
For clarify, I would name this `def is_string_template(self)` as that's what the property is really reporting.  It would also, I think (but open to persuasion!) be simpler as:

```python
    @property
    def is_string_template(self):
        """:returns: Boolean if this instance is a template initialised with a string"""
        return self._config_template is not None
```

Reply:
Agreed.